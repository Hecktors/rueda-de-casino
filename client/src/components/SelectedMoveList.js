import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import { YoutubeIcon } from './Icons'

SelectedMoveList.propTypes = {
  moves: PropTypes.array.isRequired,
  onClick: PropTypes.func,
}

export default function SelectedMoveList({ moves, onClick }) {
  const listItems = moves.map(({ name, id, videoId, videoStart }) => (
    <li key={id}>
      <button
        data-testid="listitem-button"
        onClick={() => onClick({ id: videoId, start: videoStart })}
      >
        {name}
        {videoId && (
          <span>
            <YoutubeIcon />
          </span>
        )}
      </button>
    </li>
  ))

  return (
    <SelectedMoveListStyled>
      <ul>{listItems}</ul>
    </SelectedMoveListStyled>
  )
}

const SelectedMoveListStyled = styled.div`
  width: 100%;
  max-width: 400px;
  margin: 0 auto;
  padding: 0 30px;

  ul {
    padding: 0;

    li {
      button {
        line-height: 1.4;
        width: 100%;
        font-size: 1.3rem;
        color: var(--color-listitem);
        display: flex;
        justify-content: space-between;
      }
    }

    span:hover {
      opacity: 0.6;
    }
    span {
      display: grid;
      place-items: center;
    }
  }
`
