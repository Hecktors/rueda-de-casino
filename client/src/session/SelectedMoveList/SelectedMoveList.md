```jsx
const moves = [
  {
    id: 'm1',
    name: 'la prima',
    bars: 2,
    filename: 'la_prima.mp3',
    videoUrl: 'test',
    videoStart: '20',
  },
  {
    id: 'm2',
    name: 'sombrero',
    bars: 2,
    filename: 'sombrero.mp3',
    videoUrl: 'test',
    videoStart: '1',
  },
  {
    id: 'm3',
    name: 'sombrero con una mano',
    bars: 2,
    filename: 'sombrero_con_una_mano.mp3',
    videoUrl: 'test',
    videoStart: '1',
  },
  {
    id: 'm4',
    name: 'doble play',
    bars: 3,
    filename: 'doble_play.mp3',
    videoUrl: 'test',
    videoStart: '1',
  },
  {
    id: 'm5',
    name: 'setenta',
    bars: 4,
    filename: 'setenta.mp3',
    videoUrl: 'test',
    videoStart: '1',
  },
  {
    id: 'm6',
    name: 'm√©tele el dedo',
    bars: 4,
    filename: 'metele_el_dedo.mp3',
    videoUrl: '',
    videoStart: '',
  },
]
;<div>
  <SelectedMoveList moves={moves} onClick={() => {}} />
</div>
```
