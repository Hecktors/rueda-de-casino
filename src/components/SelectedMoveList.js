import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import youTubeIcon from '../assets/img/youTube.png'

SelectedMoveList.propTypes = {
  moves: PropTypes.array.isRequired,
  onClick: PropTypes.func,
}

export default function SelectedMoveList({ moves, onClick }) {
  const listItems = moves.map(({ name, id, videoId, videoStart }) => (
    <li key={id}>
      <button
        data-testid="listitem-button"
        onClick={() => onClick({ id: videoId, start: videoStart })}
      >
        {name}
        {videoId && (
          <span>
            <img src={youTubeIcon} alt="youtube" />
          </span>
        )}
      </button>
    </li>
  ))

  return (
    <SelectedMoveListStyled>
      <ul>{listItems}</ul>
    </SelectedMoveListStyled>
  )
}

const SelectedMoveListStyled = styled.div`
  width: 100%;
  padding: 30px;

  ul {
    padding: 0;

    li {
      button {
        line-height: 1.4;
        width: 100%;
        font-size: 1.3rem;
        color: var(--color-listitem);
        display: flex;
        justify-content: space-between;
      }
    }

    span:hover {
      opacity: 0.6;
    }

    img {
      height: 14px;
    }
  }
`
