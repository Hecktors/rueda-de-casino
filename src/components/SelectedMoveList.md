```jsx
import moves from '../data/pensum.json'
;<div>
  <SelectedMoveList moves={moves} isPaused={false} />
</div>
```
