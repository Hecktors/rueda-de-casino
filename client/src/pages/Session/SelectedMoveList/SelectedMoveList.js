import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import { YoutubeIcon, YoutubeGreyIcon } from '../../../components/Icons'

SelectedMoveList.propTypes = {
  moves: PropTypes.array.isRequired,
  onClick: PropTypes.func,
}

export default function SelectedMoveList({ moves, onClick }) {
  return (
    <SelectedMoveListStyled>
      {moves.map(({ name, _id, videoUrl, videoStart }) => (
        <li key={_id}>
          <button onClick={() => onClick({ url: videoUrl, start: videoStart })}>
            <span>
              {videoUrl ? (
                <YoutubeIcon title="youtube icon" />
              ) : (
                <YoutubeGreyIcon title="gray youtube icon" />
              )}
            </span>
            {name}
          </button>
        </li>
      ))}
    </SelectedMoveListStyled>
  )
}

const SelectedMoveListStyled = styled.ul`
  width: 100%;
  max-width: 400px;
  display: grid;
  gap: 10px;
  margin: 0 auto;
  padding: 120px 0;

  @media (orientation: landscape) {
    padding: 100px 0;
  }

  li {
    padding-left: 24px;

    button {
      display: flex;
      font-size: 1.5rem;
      color: var(--color-text);

      svg {
        margin-right: 10px;
        margin-top: 3px;
      }
    }
  }
`
