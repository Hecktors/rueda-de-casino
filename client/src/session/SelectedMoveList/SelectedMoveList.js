import PropTypes from 'prop-types'
import styled from 'styled-components/macro'
import { YoutubeIcon, YoutubeGreyIcon } from '../../app/components/Icons'

SelectedMoveList.propTypes = {
  moves: PropTypes.array.isRequired,
  onClick: PropTypes.func,
}

export default function SelectedMoveList({ moves, onClick }) {
  return (
    <SelectedMoveListStyled>
      {moves.map(({ name, _id, videoUrl, videoStart }) => (
        <li key={_id}>
          <button
            data-testid="listitem-button"
            onClick={() => onClick({ url: videoUrl, start: videoStart })}
          >
            <span>{videoUrl ? <YoutubeIcon /> : <YoutubeGreyIcon />}</span>
            {name}
          </button>
        </li>
      ))}
    </SelectedMoveListStyled>
  )
}

const SelectedMoveListStyled = styled.ul`
  width: 100%;
  max-width: 400px;
  margin: 0 auto;
  padding: 120px 0;
  display: grid;
  gap: 10px;

  @media (orientation: landscape) {
    padding: 100px 0;
  }

  li {
    padding-left: 24px;
    button {
      font-size: 1.5rem;
      display: flex;
      color: var(--color-text);

      svg {
        margin-right: 10px;
        margin-top: 3px;
      }
    }
  }
`
