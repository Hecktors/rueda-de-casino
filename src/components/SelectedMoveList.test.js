import SelectedMoveList from './SelectedMoveList'
import { render } from '@testing-library/react'
const testMoves = [
  {
    id: 1,
    name: 'la prima',
  },
  {
    id: 2,
    name: 'sombrero',
  },
]

describe('SelectedMoveList', () => {
  it('renders SelectedMoveList', () => {
    const { container } = render(<SelectedMoveList moves={testMoves} />)
    expect(container).toMatchSnapshot()
  })

  it('shows move names', () => {
    const { getByText } = render(<SelectedMoveList moves={testMoves} />)
    expect(getByText('la prima')).toBeInTheDocument()
    expect(getByText('sombrero')).toBeInTheDocument()
  })
})
