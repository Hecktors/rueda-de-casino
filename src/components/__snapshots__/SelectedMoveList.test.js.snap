// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectedMoveList renders SelectedMoveList 1`] = `
<div>
  <div
    class="SelectedMoveList__SelectedMoveListStyled-sc-1k4i04h-0 dkcHxq"
  >
    <ul>
      <li>
        <button
          data-testid="listitem-button"
        >
          la prima
        </button>
      </li>
      <li>
        <button
          data-testid="listitem-button"
        >
          sombrero
        </button>
      </li>
    </ul>
  </div>
</div>
`;
